### PR TITLE
Generic trash types mod

### PR DESCRIPTION
**Pull request by SuddenEyePuncture**
#### Summary
Mods "Replace plastic and steel trash in houses with their generic counterparts"

#### Purpose of change

As suggested in this https://github.com/CleverRaven/Cataclysm-DDA/issues/56511 issue, I've created a small mod to replace all forks, combs, etc with plastic and steel junk.

#### Describe the solution

Generic trash replaces useless loot with uniform junk. You can disassemble or smelt it to get scrap metal and plastic chunks respectively. It should help new players to set their scavenging priorities. Later I'm going to add other types of trash.

#### Describe alternatives you've considered

Differentiate junk into more types, such as metal containers, plastic cintainers, etc.

#### Testing

Generated a new world, had a few errors. Game tries to put tomatoes or nachos into plastic trash and fails to do so. I've checked junk spawn in houses and it looks like it works as intended. You can smelt and disassemble junk as intended.

#### Additional context

image

image

### CODE REVIEW

**Comment by Shodan14** [comment]
Honestly, this seems pretty great for the main game as well.

File: data/mods/Generic_Trash/generic_trash.json [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    "symbol": "$",
+    "price": 2000000,
+    "price_postapoc": 0
+  },
+  {
+    "type": "GENERIC",
+    "id": "generic_wood",
+    "symbol": "/",
+    "color": "brown",
+    "name": { "str": "wooden hardware" },
```
Comment (dseguin) [inline data/mods/Generic_Trash/generic_trash.json:112]:
```suggestion
    "name": { "str_sp": "wooden hardware" },
```

File: data/mods/Generic_Trash/generic_trash.json [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    "price": 2000000,
+    "price_postapoc": 0
+  },
+  {
+    "type": "GENERIC",
+    "id": "generic_wood",
+    "symbol": "/",
+    "color": "brown",
+    "name": { "str_sp": "wooden hardware" },
+    "description": "Some kind of wooden tool or appliance. Whatever it is, you can break it into splinters.",
```
Comment (dseguin) [inline data/mods/Generic_Trash/generic_trash.json:113]:
```suggestion
    "description": "Some kind of wooden tool or appliance.  Whatever it is, you can break it into splinters.",
```

**Review by dseguin — COMMENTED** [review]
Just some more translation conventions, but also you would probably need to specify a behaviour for `"on_overflow"` for the containers to fix the test errors (like this one: https://github.com/CleverRaven/Cataclysm-DDA/runs/6251197162#step:16:7188)

**Comment by SuddenEyePuncture** [comment]
Thank you very much for your review! I'll fix overflow issue later, when I'll have access to my main computer.

File: data/mods/Generic_Trash/generic_trash.json [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    "symbol": ",",
+    "color": "light_gray"
+  },
+  {
+    "type": "GENERIC",
+    "id": "generic_ceramic",
+    "symbol": "*",
+    "color": "white",
+    "name": { "str": "ceramic dish", "str_pl": "ceramic dishes" },
+    "description": "Ceramic cups, plates and bowls.  Whatever it is, you can brake it into a sharp shard.",
```
Comment (Night-Pryanik) [inline data/mods/Generic_Trash/generic_trash.json:64]:
```suggestion
    "description": "Ceramic cups, plates and bowls.  Whatever it is, you can break it into a sharp shard.",
```

File: data/mods/Generic_Trash/generic_trash.json [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+    "color": "green",
+    "symbol": "$",
+    "price": 200000,
+    "price_postapoc": 0
+  },
+  {
+    "type": "GENERIC",
+    "id": "generic_money_bundle",
+    "name": { "str": "currency bundle" },
+    "description": "A thousand pre_Cataclysm bills, divided evenly between ten currency straps, and bundled together with two rubber bands.",
```
Comment (Night-Pryanik) [inline data/mods/Generic_Trash/generic_trash.json:126]:
```suggestion
    "description": "A thousand pre-Cataclysm bills, divided evenly between ten currency straps, and bundled together with two rubber bands.",
```

File: data/mods/Generic_Trash/generic_trash.json [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+    "price_postapoc": 0,
+    "material": [ "paper" ],
+    "volume": "6 ml",
+    "longest_side": "10 mm",
+    "ammo_type": "paper"
+  },
+  {
+    "id": "generic_glass",
+    "name": { "str": "glass dish", "str_pl": "glass dishes" },
+    "description": "Glass cups, plates and bowls.  Whatever it is, you can brake it into sharp shards.",
```
Comment (Night-Pryanik) [inline data/mods/Generic_Trash/generic_trash.json:168]:
```suggestion
    "description": "Glass cups, plates and bowls.  Whatever it is, you can break it into sharp shards.",
```

**Comment by I-am-Erk** [comment]
The concept described there is not to collapse junk items like this, but to blacklist them for performance reasons. I don't think this meets the mod guidelines, as it still generates all the same items, just with different names.

**Comment by kevingranade** [comment]
The original premise was also flawed, I would not expect this to have any impact on performance.

**Pull request closed without merging**